### PR TITLE
Add vault job for pulp2

### PR DESCRIPTION
**Pull request by rochacbruno**
This job will:
 - Provision a new machine on OpenStack 
 - Install INITIAL_PULP_VERSION (usually one version before the latest). 
 - Install pulp-smash from SMASH_REPO.
- Clone pulp-2-testsfrom TEST_REPO.
- Run py.test sanity check using args provided in PRE_PYTEST_ARGS.
- If it passes then it will upgrade O.S using OS_LATEST_REPO.
- Reboot the instance
- Then it will run the tests again this time using POST_PYTEST_ARGS.

### CODE REVIEW

File: ci/jjb/scripts/fix_remote_hostname.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env bash
+# This information can aid in debugging failed jobs.
+ssh cloud-user@${PULP_HOSTNAME} << EOF
+hostname --all-fqdn
+hostname
+# Setting the hostname of the system to enable connectivity
+hostname="$(hostname --all-fqdn | grep openstacklocal | head -n 1)"
```
Comment (bherrin3) [inline ci/jjb/scripts/fix_remote_hostname.sh]:
```suggestion
hostname="$(hostname --all-fqdn | grep upshift | head -n 1)"
```

File: ci/jjb/jobs/pulp-vault.yaml [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+           default: pulp-jenkins
+           description: Can be found on OPenstack user interface.
+        - string:
+           name: UPSHIFTAUTHURL
+           default: https://rhos-d.infra.prod.upshift.rdu2.redhat.com:13000/v3
+           description: Can be found on OPenstack user interface.
+
+        - string:
+            name: PULP_VERSION
+            default: '2.19'
```
Comment (bherrin3) [inline ci/jjb/jobs/pulp-vault.yaml]:
```suggestion
            default: '2.21'
```

:)

File: ci/jjb/jobs/pulp-vault.yaml [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+        - shell: |
+            # Useful for debugging
+            getent passwd "$(id -u)"
+
+            # Setup ssh config and private key
+            mkdir ~/.ssh/controlmasters
+            
+            cat > ~/.ssh/config <<EOF
+            Host ${PULP_HOSTNAME}
+                User cloud-user
```
Comment (bherrin3) [inline ci/jjb/jobs/pulp-vault.yaml]: The `cloud-user` was just for RHEL. Fedora will use `fedora`. It will be different depending on the base template. :(

Comment (rochacbruno) [reply ci/jjb/jobs/pulp-vault.yaml]: This vault job will always run RHEL

File: ci/jjb/jobs/macros.yaml [hunk 4]
```diff
@@ -56,7 +56,7 @@
             hostname --all-fqdn
             hostname
             # Setting the hostname of the system to enable connectivity
-            hostname="$(hostname --all-fqdn | grep openstacklocal | head -n 1)"
+            hostname="$(hostname --all-fqdn | grep upshift | head -n 1)"
```
Comment (bherrin3) [inline ci/jjb/jobs/macros.yaml]: @rochacbruno I think if you rebase against `production master`, this has already been merged in.

**Comment by bherrin3** [comment]
Had a quick 1v1 to let @rochacbruno know about DNS problems with the ENV with the current backwards compatible work-around in #682 .

Otherwise, +1 !

**Comment by ggainey** [comment]
This would prob have been useful - but pulp2 is EOL now, so I'm going to close the PR. Thanks for the contribution, sorry it never made it into production :(

**Pull request closed without merging**